modules/collator-selection/src/lib.rs

```diff
@@ -406,30 +397,83 @@ pub mod pallet {
 	/// Play the role of the session manager.
 	impl<T: Config> SessionManager<T::AccountId> for Pallet<T> {
 		fn new_session(index: SessionIndex) -> Option<Vec<T::AccountId>> {
-			log::info!(
-				"assembling new collators for new session {} at #{:?}",
+			let candidates = Self::candidates()
+				.iter()
+				.map(|candidate| candidate.who.clone())
```

Comment:
```suggestion
				.into_iter()
				.map(|candidate| candidate.who)
```

I think this can avoid the clone